docker/dev/find_fastest_resources.sh

```diff
@@ -24,139 +24,207 @@
+    read -r -d '' apt_sources <<- EOM
+http://mirrors.aliyun.com
+http://mirrors.ustc.edu.cn
+http://mirrors.163.com
+http://archive.ubuntu.com
+EOM
+
+    # Find the fastest APT source using ping.
+    local fastest
+    # shellcheck disable=SC2086
+    fastest=$(find_fastest_url $apt_sources)/ubuntu/
+
+    # The default Ubuntu version is 18.04, code named bionic.
+    local codename
+    codename=${ubuntu_codename-"bionic"}
+
+    # Write APT source lists.
+    cat <<-EOF
+deb $fastest $codename main restricted universe multiverse
+deb $fastest $codename-security main restricted universe multiverse
+deb $fastest $codename-updates main restricted universe multiverse
+deb $fastest $codename-proposed main restricted universe multiverse
+deb $fastest $codename-backports main restricted universe multiverse
+deb-src $fastest $codename main restricted universe multiverse
+deb-src $fastest $codename-security main restricted universe multiverse
+deb-src $fastest $codename-updates main restricted universe multiverse
+deb-src $fastest $codename-proposed main restricted universe multiverse
+deb-src $fastest $codename-backports main restricted universe multiverse
+EOF
 }
 
 function find_fastest_maven_repo() {
-	# c.f. https://unix.stackexchange.com/questions/353076
-	# we can indent EOM wiht tab if we use the "<<-" form
-	read -r -d '' maven_repos <<-EOM
-		https://repo1.maven.org/maven2/
-		https://maven.aliyun.com/repository/central
-	EOM
-
-	local best_maven_repo=$(find_fastest_url ${maven_repos})
-	local domain=$(get_domain_from_url $best_maven_repo)
-
-	cat <<-EOF
-	<?xml version="1.0" encoding="UTF-8"?>
-	<settings xmlns="http://maven.apache.org/SETTINGS/1.0.0"
-	    xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
-	    xsi:schemaLocation="http://maven.apache.org/SETTINGS/1.0.0 http://maven.apache.org/xsd/settings-1.0.0.xsd">
-	    <mirrors>
-	        <mirror>
-	            <id>${domain}</id>
-	            <mirrorOf>central</mirrorOf>
-	            <name>${best_maven_repo}</name>
-	            <url>${domain}</url>
-	        </mirror>
-	    </mirrors>
-	</settings>
-	EOF
+    # c.f. https://unix.stackexchange.com/questions/353076
+    # we can indent EOM wiht tab if we use the "<<-" form
+    #
+    read -r -d '' maven_repos <<-EOM
+https://repo1.maven.org/maven2/
+https://maven.aliyun.com/repository/central
+EOM
+
+    local best_maven_repo
+    # shellcheck disable=SC2086
+    best_maven_repo=$(find_fastest_url $maven_repos)
+    local domain
+    domain=$(get_domain_from_url "$best_maven_repo")
+
+    cat <<-EOF
+<?xml version="1.0" encoding="UTF-8"?>
+<settings xmlns="http://maven.apache.org/SETTINGS/1.0.0"
+    xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+    xsi:schemaLocation="http://maven.apache.org/SETTINGS/1.0.0 http://maven.apache.org/xsd/settings-1.0.0.xsd">
+    <mirrors>
+        <mirror>
+            <id>$domain</id>
+            <mirrorOf>central</mirrorOf>
+            <name>$best_maven_repo</name>
```

Comment:
<name>$best_maven_repo</name>  -> <name>$domain</name>
<url>$domain</url>  -> <url>$best_maven_repo</url>

sorry, this is a previous bug.